java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1651 @@
+        debugInstrument.log("Exit at line %d%n", line_number);
+
+        // only do incremental lines if we don't have the line generator
+        if (line_number == last_line_number && foundLine == false) {
+          debugInstrument.log("Could not find line... at %d%n", line_number);
+          line_number++;
+        }
+
+        last_line_number = line_number;
+
+        if (!shouldIgnore(
+            classInfo.class_name,
+            mgen.getName(),
+            DaikonWriter.methodExitName(
+                classInfo.class_name,
+                getFullyQualifiedArgTypeNames(mgen),
+                mgen.toString(),
+                mgen.getName(),
+                line_number))) {
+          shouldInclude = true;
+          exit_locs.add(line_number);
+
+          isIncluded.add(true);
+        } else {
+          isIncluded.add(false);
+        }
+      }
+    }
+
+    if (shouldInclude) {
+      return new MethodInfo(
+          classInfo, mgen.getName(), arg_names, arg_type_strings, exit_locs, isIncluded);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Returns true if the specified class is part of Chicory itself (and thus should not be
+   * instrumented). Some Daikon classes that are used by Chicory are included here as well.
+   *
+   * @param classname the name of the class to test, in internal form
+   * @return true if the given class is part of Chicory itself
+   */
+  @Pure
+  private static boolean is_chicory(@InternalForm String classname) {
+
+    if (classname.startsWith("daikon/chicory") && !classname.equals("daikon/chicory/ChicoryTest")) {
+      return true;
+    }
+    if (classname.equals("daikon/PptTopLevel$PptType")) {
+      return true;
+    }
+    if (classname.startsWith("daikon/plumelib")) {
+      return true;
+    }
+    return false;
+  }
+
+  // UNFINISHED and maybe unneeded
+  //  // converts a method descriptor to a Java language string
+  //  public static String convertDescriptorToString(String descriptor) {
+  //    StringBuilder args = new StringBuilder("(");
+  //    if (descriptor.charAt(0) != '(') {
+  //        throw new IllegalArgumentException("Invalid method descriptor: " + descriptor);
+  //    }
+  //    int paren = descriptor.indexOf(')');
+  //    if (paren < 0) {
+  //        throw new IllegalArgumentException("Invalid method descriptor: " + descriptor);
+  //    }
+  //    int end;
+  //    int comma = descriptor.indexOf(',');
+  //    if (comma < 0) {
+  //        end = paren;
+  //    } else {
+  //        end = min(comma, paren);
+  //    }
+
+  /**
+   * Format a field descriptor for output. The main difference between a descriptor and a signature
+   * is that the later may contain type arguments. This routine was orginaly written for
+   * descriptors, but some support for type arguments has been added.
+   *
+   * @param descriptor the object to format
+   * @return a formatted string
+   */
+  public static String convertDescriptorToString(String descriptor) {
```

Comment:
Please rename this from "...ToString" to indicate what the output format is.  Also, describe that in the Javadoc.